### PR TITLE
Add map embeds

### PR DESCRIPTION
**Pull request by Pixelguymm**

### CODE REVIEW

File: src/jsMain/kotlin/io/beatmaps/maps/mapEmbed.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        val mapKey = params["mapKey"]
+
+        setMap(null)
+
+        axiosGet<MapDetail>(
+            "${Config.apibase}/maps/id/$mapKey"
+        ).then {
+            setMap(it.data)
+        }.catch {
+            history.push("/")
```
Comment (Top-Cat) [inline src/jsMain/kotlin/io/beatmaps/maps/mapEmbed.kt:34]: I really don't think you want to do this. Need some fancy error message instead.

The header isn't there and it's going to be weird having the homepage in an iframe...
Screenshot 2024-06-24 at 15 31 16

**Comment by Top-Cat** [comment]
Should hide the add to playlist button in the embed as it requires the modal popup

**Comment by Pixelguymm** [comment]
It didn't appear on foreign sites while testing, even when logged in. Did you get it to appear?

**Comment by Top-Cat** [comment]
> It didn't appear on foreign sites while testing, even when logged in. Did you get it to appear?

Ah yes, I just went to `/embed` but of course the cookies won't work when actually in an iframe.

File: src/jsMain/kotlin/io/beatmaps/maps/mapEmbed.kt (hunk 1 above)
Comment (Pixelguymm) [reply src/jsMain/kotlin/io/beatmaps/maps/mapEmbed.kt:34]: Oh you're right, I'll see what I can do to fix that.

**Pull request merged**